pkg/controller/combo_controller.go

```diff
@@ -2,36 +2,134 @@ package controller
 
 import (
 	"context"
+	"fmt"
+	"strings"
+	"time"
 
 	"github.com/go-logr/logr"
+	"github.com/operator-framework/combo/api/v1alpha1"
+	combinationPkg "github.com/operator-framework/combo/pkg/combination"
+	comboConditions "github.com/operator-framework/combo/pkg/conditions"
+	templatePkg "github.com/operator-framework/combo/pkg/template"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
 
-	"github.com/operator-framework/combo/api/v1alpha1"
+var (
+	RequeueDefaultTime = time.Second * 2
+	SuccessCondition   = comboConditions.ProccessedCombinationsCondition
 )
 
 type combinationController struct {
 	client.Client
 	log logr.Logger
 }
 
+// manageWith creates a new instance of this controller
 func (c *combinationController) manageWith(mgr ctrl.Manager, version int) error {
 	c.log = c.log.V(version)
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&v1alpha1.Combination{}).
 		Complete(c)
 }
 
-func (c *combinationController) Reconcile(ctx context.Context, req ctrl.Request) (reconcile.Result, error) {
-	// Set up a convenient log object so we don't have to type request over and over again
+// Reconcile manages incoming combination CR's and processes them accordingly
+func (c *combinationController) Reconcile(ctx context.Context, req ctrl.Request) (result reconcile.Result, deferredErr error) {
+	// Set up a convenient log object so we don’t have to type request over and over again
 	log := c.log.WithValues("request", req)
-	log.Info("reconciling combination")
 
-	in := &v1alpha1.Combination{}
-	if err := c.Get(ctx, req.NamespacedName, in); err != nil {
+	log.Info("new combination event inbound")
```

Comment:
nit: We'll want to bump the log verbosity for this kind of log.

Reply:
What level of verbosity were you thinking?

Reply:
Went ahead with level 2 to get it out of the default level